### ISSUE TITLE
Loading thumbs and corresponding Full size images with pagination

### ISSUE THREAD

**Issue by gabbimark** [bumptech/glide#1586]
Hi we are trying to use Glide in our project to load images to surface view. We download a list of thumbs and Full size Pics together.  Here is what we are trying to do

See attached picture for reference
1. We want to load list of thumbs from the server. Users can swipe left to right to load up more thumbs into the view. We do not load all thumbs at once. We only want to load 4 thumbs at a time from server using pagination principle as user scrolls to the right. For example if there are 20 images for the user to view pending on the server. When the users accesses the page we load up the first 4. When user scrolls right we load up next 4. 

2. When thumbs are loaded corresponding Full size images need to get loaded from server as well. 

3. Users can view full size images in two ways. (1) click on any corresponding thumbnail to load up any image out of sequence or in no particular order (2) swipe left or right on the full size image to load up next image in sequential manner. 

Please refer to the reference image i uploaded. How do we accomplish the above 3 tasks with glide if possible? Please give us the sample code to perform these tasks as we do not see how to do this in docs. 
![mediapage](https://cloud.githubusercontent.com/assets/10815053/20250687/62ada93e-a9df-11e6-9d69-54a6d3942f64.jpg)

**Comment by TWiStErRob** [comment]
Most of what you're asking is out of scope/independent of Glide. If we assume you have the thumbnail urls/full image urls, the solution is just to simply call a `Glide.....load.into` whenever the user scrolls (simplest to use List/RecyclerView).

The kind of batching you want is not supported by any library as far as I know, if you want to load 4 thumbs at once (in a ZIP file?) you need to do that manually, and then use Glide to load the downloaded `File`. But if you don't have ZIPs or other strange delivery mechanism, you can use the `ListPreloader` to load images that are off-screen in a list to make it look like the user has infinitely fast internet/you downloaded everything in advance.
